pio/i2c/i2c.pio

```diff
@@ -108,10 +108,11 @@ static inline void i2c_program_init(PIO pio, uint sm, uint offset, uint pin_sda,
     gpio_set_oeover(pin_scl, GPIO_OVERRIDE_INVERT);
     pio_sm_set_pins_with_mask(pio, sm, 0, both_pins);
 
-    // Clear IRQ flag before starting
-    hw_clear_bits(&pio->inte0, 1u << sm);
-    hw_clear_bits(&pio->inte1, 1u << sm);
-    pio->irq = 1u << sm;
+    // Clear IRQ flag before starting, and make sure flag doesn't actually
+    // assert a system-level interrupt (we're using it as a status flag)
+    pio_set_irq0_source_mask_enabled(pio, 1u << sm, false);
```

Comment:
ah, that is what confused me 1u << sm is NOT the enable for the IRQ; it is one of the FIFO statuses...
perhaps use `pio_set_irq0_source_enabled(pio, pis_interrupt0 + sm, false);`

Reply:
Oops